### PR TITLE
Speaking Club

### PR DESCRIPTION
**Pull request by LGretzk**
## Description

- Describe your changes in detail.
Add JSON page for the Speaking Club Programme.
Add test for the ProgramType.

- Why is this change required?
Create and Test API for the SpeakingClubPage.json file (resources)

- What problem does it solve?
Data available in page response.

- Add Notes if anything is left unclear or not completed
The page still need more data to be complete, i.e., image links if required.

## Related Issue

Closes: #334 

## Change Type

- [ ] Bug Fix
- [ ] New Feature
- [ ] Code Refactor
- [ ] Documentation
- [ ] Test
- [ ] Other

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] I checked and followed the [contributor guide](../CONTRIBUTING.md)
- [ ] I have tested my changes locally.

### CODE REVIEW

File: src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+/** Test class for {@link ProgramType}. */
+class ProgramTypeTest {
+
+  @Test
+  void shouldReturnCorrectDescriptionForEnumValues() {
+    // Given: All enum values in ProgramType
```
Comment (nverbic) [inline src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java:12]: Please use this test implementation [here ](https://github.com/Women-Coding-Community/wcc-backend/blob/main/src/test/java/com/wcc/platform/domain/cms/attributes/style/ColorTypeTest.java#L11-L15) since it using some better approach with parametrized option that will automatically create test for each enum param.

File: src/main/resources/speakingClubPage.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
...iver your talks and presentations to a supportive audience. Get immediate feedback and constructive criticism to help you improve.\nSpeech Evaluations and Feedback: Receive detailed evaluations from peers and mentors to refine your delivery and content. Use personalised feedback to enhance your strengths and address areas for improvement.\nGuest Speaker Events: Gain insights from guest speakers who are experts in public speaking, presentation skills, and communication. Learn best practices and tips from successful orators and industry leaders.\nOnline Debates and Discussions: Engage in structured online debates to sharpen your argumentation and persuasive speaking skills. Participate in group discussions on a variety of topics to improve your ability to articulate ideas clearly.\nWebinars and Q&A Sessions: Attend webinars covering various aspects of public speaking and presentation techniques. Participate in Q&A sessions with experienced speakers to get your questions answered.\nNetworking and Community:Connect with like-minded individuals in our virtual networking events and slack channel Build a network of friends, mentors, and collaborators who share your passion for speaking.",
```
Comment (nverbic) [inline src/main/resources/speakingClubPage.json:28]: In another PR you can do the following: 
 - Please remove the "card": null below. 
 - Check all other programmes' json files (also in test folder under resources) and remove "card": null entries if any  (like in interview prep json for example).
 - Remove the @NotNull annotation from the [file](https://github.com/Women-Coding-Community/wcc-backend/blob/main/src/main/java/com/wcc/platform/domain/platform/Programme.java#L23) 
 - Run tests and start the application and test the API in Swagger to insure that the programmes pages work as expected.

File: src/main/resources/speakingClubPage.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  "id": "page:SPEAKING_CLUB",
+  "heroSection": {
+    "title": "WWC Speaking Club"
+  },
+  "images": [
+    {
+      "path": "",
+      "alt": "",
+      "type": ""
+    }
```
Comment (nverbic) [inline src/main/resources/speakingClubPage.json:11]: Please use the same as in interview prep page [here](https://github.com/Women-Coding-Community/wcc-backend/blob/main/src/main/resources/interviewPreparationPage.json#L7-L9). We will later update to some other image (we do not have it yet).

File: src/main/resources/speakingClubPage.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "links": [
+      {
+        "type": "SLACK",
+        "link": "http://shortlink_to_slack.com/#prog_speaking_club"
+      }
+    ]
+  },
+  "programmeDetails": [
+    {
+      "title": "What we offer",
```
Comment (nverbic) [inline src/main/resources/speakingClubPage.json:27]: In the description below please remove all new lines \n and leave text as just one  block. We will format later.

File: src/main/resources/speakingClubPage.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  ],
+  "section": {
+    "description": "Are you looking to improve your public speaking skills, enhance your presentation abilities, and gain confidence in delivering talks? The Online Speaking Club is your ultimate platform to practise and refine your speaking skills in a supportive and interactive virtual environment. Whether you’re preparing for a professional presentation, a public talk, or simply want to improve your communication skills, our club offers the resources and community you need to succeed."
+  },
+  "contact": {
+    "title": "Join us in our Speaking Club Slack channel",
+    "links": [
+      {
+        "type": "SLACK",
+        "link": "http://shortlink_to_slack.com/#prog_speaking_club"
```
Comment (nverbic) [inline src/main/resources/speakingClubPage.json:21]: You can use the actual Slack link here: https://womencodingcommunity.slack.com/archives/C070NACTMCM

File: src/main/resources/speakingClubPage.json (hunk 2 above)
Comment (nverbic) [reply src/main/resources/speakingClubPage.json:28]: We will crate an issue with the above explanation first and then assign it to you.

**Review by nverbic — APPROVED** [review]
Great! Approved

File: src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    ProgramType[] programTypes = ProgramType.values();
+
```
Comment (nverbic) [inline src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java:14]: You can remove this variable since it is not used.

File: src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java (hunk 1 above)
Comment (LGretzk) [reply src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java:12]: Yes, adding new programmes to the test is not ideal... Unfortunately, the structure of the `ColorType` enum is a bit different, here descriptions are derived from the `enum` names and `@EnumSource` and can be tested in a clean way. 
`ProgramType` descriptions are custom we need to store the expected values somewhere. I wonder if it's worth keeping the test at all?

File: src/main/resources/speakingClubPage.json (hunk 4 above)
Comment (LGretzk) [reply src/main/resources/speakingClubPage.json:27]: done 🙌

File: src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java (hunk 1 above)
Comment (LGretzk) [reply src/test/java/com/wcc/platform/domain/platform/ProgramTypeTest.java]: Thanks, I've replaced the test.
@dricazenck done! Can I merge it?
